### PR TITLE
issue #144 - spread remaining props on parent div

### PR DESCRIPTION
**Pull request by cklab**

### CODE REVIEW

**Comment by cklab** [comment]
Rebased

**Comment by cklab** [comment]
Rebased.

**Comment by ad1992** [comment]
@cklab can you rebase with latest master?

**Comment by cklab** [comment]
@ad1992 rebased

File: lib/ReactTags.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import PropTypes from "prop-types";
 import Tag from "./Tag";
 
 //Constants
-import { KEYS, DEFAULT_PLACEHOLDER, DEFAULT_CLASSNAMES } from './constants';
+import { KEYS, DEFAULT_PLACEHOLDER, DEFAULT_CLASSNAMES } from "./constants";
```
Comment (ad1992) [inline lib/ReactTags.js]: single quotes ?

File: lib/ReactTags.js [hunk 2]
```diff
@@ -308,11 +315,16 @@ class ReactTags extends Component {
...
+      suggestions = this.state.suggestions;
+
+    let {
+      placeholder,
+      name: inputName,
+      id: inputId,
+      maxLength,
+      ...otherProps
+    } = this.props;
+    otherProps = this.removeInternalProps(otherProps);
```
Comment (ad1992) [inline lib/ReactTags.js]: lets try not using `let` and do 
`
const modifiedProps = this.removeInternalProps(rest);
`

File: lib/ReactTags.js [hunk 3]
```diff
@@ -308,11 +315,16 @@ class ReactTags extends Component {
...
-      inputId = this.props.id,
-      maxLength = this.props.maxLength;
+      suggestions = this.state.suggestions;
+
+    let {
+      placeholder,
+      name: inputName,
+      id: inputId,
+      maxLength,
+      ...otherProps
```
Comment (ad1992) [inline lib/ReactTags.js]: can you rename otherProps to `rest` ?

File: test/reactTags.test.js [hunk 4]
```diff
@@ -201,6 +201,21 @@ test("should not set any property of this.textInput when readOnly", () => {
   $tag.simulate("click");
 });
 
+test("accepts non-internal props on the top-level div", () => {
```
Comment (ad1992) [inline test/reactTags.test.js]: should accept non-internal props on the top-level div ?

File: lib/ReactTags.js [hunk 5]
```diff
@@ -389,6 +400,7 @@ ReactTags.propTypes = {
   id: PropTypes.string,
   maxLength: PropTypes.string,
   inputValue: PropTypes.string,
+  tags: PropTypes.array,
```
Comment (ad1992) [inline lib/ReactTags.js]: PropTypes.array.isRequired ?

File: lib/ReactTags.js (hunk 2 above)
Comment (ad1992) [inline lib/ReactTags.js]: why do we need this method `this.removeInternalProps` ?

File: lib/ReactTags.js (hunk 1 above)
Comment (cklab) [reply lib/ReactTags.js]: This was the output of npm run format.

File: lib/ReactTags.js (hunk 2 above)
Comment (cklab) [reply lib/ReactTags.js]: See comment https://github.com/prakhar1989/react-tags/issues/144#issuecomment-363223276

File: lib/ReactTags.js (hunk 5 above)
Comment (cklab) [reply lib/ReactTags.js]: Is it required?

File: lib/ReactTags.js (hunk 3 above)
Comment (cklab) [reply lib/ReactTags.js]: Sure.

File: lib/ReactTags.js (hunk 2 above)
Comment (cklab) [reply lib/ReactTags.js]: Ok.

File: test/reactTags.test.js (hunk 4 above)
Comment (cklab) [reply test/reactTags.test.js]: Ok.

File: lib/ReactTags.js (hunk 1 above)
Comment (ad1992) [reply lib/ReactTags.js]: @cklab i just merged a commit which has changed the configuration of prettier so please run it again.

File: lib/ReactTags.js (hunk 5 above)
Comment (ad1992) [reply lib/ReactTags.js]: yup its required thats why we have `[]` as default prop

**Comment by cklab** [comment]
Rebased once more.

File: lib/ReactTags.js (hunk 1 above)
Comment (cklab) [reply lib/ReactTags.js]: I see, done.

File: lib/ReactTags.js [hunk 6]
```diff
@@ -308,11 +313,16 @@ class ReactTags extends Component {
...
-      inputId = this.props.id,
-      maxLength = this.props.maxLength;
+      suggestions = this.state.suggestions;
+
+    const {
+      placeholder,
+      name: inputName,
+      id: inputId,
+      maxLength,
+      ...remainingProps
```
Comment (ad1992) [inline lib/ReactTags.js]: didn't we decide to rename this to `rest` ?

File: lib/ReactTags.js [hunk 7]
```diff
@@ -308,11 +313,16 @@ class ReactTags extends Component {
...
+    } = this.props;
+    const propsToSpread = this.removeInternalProps(remainingProps);
```
Comment (ad1992) [inline lib/ReactTags.js]: can this be renamed `externalProps` ?

File: test/reactTags.test.js [hunk 8]
```diff
@@ -201,6 +201,21 @@ test('should not set any property of this.textInput when readOnly', () => {
   $tag.simulate('click');
 });
 
+test('should accept non-internal props on the top-level div', () => {
+  var onMouseOutCalled = false;
```
Comment (ad1992) [inline test/reactTags.test.js]: let ?

**Comment by ad1992** [comment]
@cklab can you rebase with latest master and make the review fixes?

**Comment by cklab** [comment]
Apologies, I haven't used this library for almost a year now. I don't have the luxury to continually rebase and resolve conflicts for trivial reasons.

**Pull request closed without merging**